tests/helpers/test_collection.py

```diff
@@ -195,10 +195,10 @@ async def test_storage_collection(hass: HomeAssistant) -> None:
     store = storage.Store(hass, 1, "test-data")
     await store.async_save(
         {
-            "items": [
-                {"id": "mock-1", "name": "Mock 1", "data": 1},
-                {"id": "mock-2", "name": "Mock 2", "data": 2},
-            ]
+            "items": {
+                "mock-1": {"name": "Mock 1", "data": 1},
```

Comment:
I am not a fan of keeping the ID out of the dictionary itself. At some point you're passing the data around and now instead of 1 item, you need to pass around 2 values, and those need to stay in sync too.

Reply:
Hmm, but `collection.py` already mostly separates the ID from the data and passes them around separately, but with the ID also present in the data.

For example, the change listeners https://github.com/home-assistant/core/blob/30204f8c197bd0895a435480e9aaf63bb00ceaca/homeassistant/helpers/collection.py#L38-L64 get the id as one parameter of the changeset, and the the data as another.

Reply:
The reason for that one was that the ID field is always dynamic, so generic helpers need to know what the ID is. However, if I were to send object X to the frotend, I would now need to wrap it in {id, data}

Reply:
That's the point of the PR basically: the storage collection should be allowed to story any items and the ID assigned to items by the storage collection should be kept separate from the items.

The frontend API is not changed in this PR, but it too should be improved IMHO since the ID is in some cases mixed with the data as an `id` key, and in some cases in a separate parameter with a different name.